### DISCUSSION TITLE
Transforming Prisma's DateTime to a usable ISO date string and back

### DISCUSSION THREAD

**Discussion by zanami** [blitz-js/blitz#2497] [Q&A]
I'm struggling with DateTime as I only need date for a DOB field, for example. I can't populate <input type=date> with what Prisma returns for a DateTime field (something like 1999-06-05T00:00:00.000Z) and I can't  save the value from that input as it is back to Prisma (it's a string like '1999-06-05').

For now i'm using zod parse/transform in queries and mutations but I don't feel it's the best (or an OK even) solution

```
...
export const UpdateProfile = Profile.extend({
  dob: z.string().transform((value) => new Date(value).toISOString()),
})

export const GetProfile = Profile.extend({
  dob: z.date().transform((value) => value.toISOString().substr(0, 10)),
})

export default async function getCurrentProfile(_ = null, { session }: Ctx) {
  if (!session.userId) return null

  const profile = await db.profile.findFirst({
    where: { userId: session.userId },
  })

  return GetProfile.parse(profile)
}
```

Am I missing something? Any better ideas? Please share!

**Comment by flybayer** [comment]
My recommendation is to store birthdates as strings in your DB. Unless you are storing exact birthdate and time like for a hospital system or something :)

**Comment by Daidalos117** [comment]
How did you solve this @zanami ? I cannot seem to make it work. Zod returns Date object, but Prisma doesn't take it, don't know why. Saving as a string doesn't seems to smart IMHO since we will loose psql builtin date functions.

**Reply by zanami** [reply]
Sorry to disappoint, but...

`dob         String? //was DateTime`

It's ok for birthdates. I actually ended up with String fields for all user provided dates. I don't need time or timezones though, just dates.
